spec/jmap/push.mdown

```diff
@@ -26,25 +31,39 @@ Refer to the Authentication section of this spec for details on how to get the U
 
 A client MAY hold open multiple connections to the event-source, although it SHOULD try to use a single connection for efficiency.
 
-## Web hook
+## Callback URL
 
-The second push mechanism is to register a callback URL to which the JMAP server will make an HTTPS POST request whenever the event occurs. The request MUST have a content type of `application/json` and contain the same UTF-8 JSON encoded object as described above as the body.
+The second push mechanism is to register a callback URL to which the JMAP server will make an HTTPS POST request whenever an event occurs. This is compatible with the Application Server role of [@!RFC8030].
 
-The JMAP server MUST also set the following headers in the POST request:
-- `X-JMAP-EventType: state`
-- `X-JMAP-User: ${username}` where `${username}` is the username of the authenticated user for which the push event occurred.
+A client MAY use the rest of [@!RFC8030] in combination with its own Push Server to form a complete solution, or MAY rely on alternative mechanisms to ensure the delivery of the pushed data after it leaves the JMAP server.
 
-The JMAP server MUST follow any redirects. If the final response code from the server is `2xx`, the callback is considered a success. If the response code is `503` (Service Unavailable), the JMAP server MAY try again later (but may also just drop the event). If the response code is `429` (Too Many Requests) the JMAP server SHOULD attempt to reduce the frequency of pushes to that URL. Any other response code MUST be considered a **permanent failure** and the callback should be deregistered (not tried again even for future events unless explicitly re-registered by the client).
+The request MUST have a content type of `application/json` and contain the same UTF-8 JSON encoded object as described above as the body.
 
-The URL set by the client MUST use the HTTPS protocol and SHOULD encode within it a unique token that can be verified by the server to know that the request comes from the JMAP server the authenticated client connected to.
+The JMAP server MUST set the following header in the POST request:
+
+- `TTL`: A value in seconds that suggests how long a push message is retained by the push service, as per [@!RFC8030] section 5.2.
+
+The JMAP server MAY include the following headers in the POST request:
+
+- `Urgency`: One of `very-low`, `low`, `normal`, or `high` as per [@!RFC8030]
+  section 5.3.
+- `Topic`: A token that encodes the set of types included in the *changed*
+  property of this push request. This MUST be restricted to no more than 32
+  characters from the URL and filename-safe Base 64 alphabet ([@!RFC4648]), as
+  per [@!RFC8030] section 5.4. This allows the push server to replace
+  undelivered messages which are entirely superceded by the later message.
```

Comment:
This just repeats what RFC 8030 says.  Are there any semantics that you would like to attach to these (I think that the answer is, and should be, no)?  If there aren't semantics, I don't see why you would mention them.